### PR TITLE
DLS-4228 Library Upgrades Play 26

### PR DESCRIPTION
**Pull request by rajanimohan22**
Library Upgrades Play 26

### CODE REVIEW

File: project/AppDependencies.scala [hunk 1]
```diff
@@ -6,22 +6,23 @@ import sbt._
...
+    "uk.gov.hmrc"                %% "domain"                     % "5.11.0-play-26",
+    "uk.gov.hmrc"                %% "play-language"              % "5.0.0-play-26",
     "org.typelevel"              %% "cats-core"                  % "2.1.0",
     "com.github.kxbmap"          %% "configs"                    % "0.4.4",
     "org.julienrf"               %% "play-json-derived-codecs"   % "7.0.0",
     "com.github.julien-truffaut" %% "monocle-core"               % "2.0.0",
     "com.github.julien-truffaut" %% "monocle-macro"              % "2.0.0",
-    "com.github.ghik"             % "silencer-lib"               % "1.6.0" % Provided cross CrossVersion.full
+    compilerPlugin("com.github.ghik" % "silencer-plugin" % "1.7.0" cross CrossVersion.full),
+    "com.github.ghik" % "silencer-lib" % "1.7.0" % Provided cross CrossVersion.full
```
Comment (rambabu-posa) [inline project/AppDependencies.scala:21]: Just suggestion, am using "1.7.1" for other services. No issue to use this

File: build.sbt [hunk 2]
```diff
@@ -15,16 +15,16 @@ resolvers += Resolver.bintrayRepo("hmrc", "releases")
 
 lazy val wartremoverSettings =
   Seq(
-    wartremoverErrors in (Compile, compile) ++= Warts.allBut(
+    (Compile / compile / wartremoverErrors) ++= Warts.allBut(
```
Comment (rambabu-posa) [inline build.sbt:18]: Use only this 
`.enablePlugins(play.sbt.PlayScala, SbtDistributablesPlugin)
`
and also remove one import

File: build.sbt [hunk 3]
```diff
@@ -49,9 +49,9 @@ lazy val microservice = Project(appName, file("."))
   .disablePlugins(JUnitXmlReportPlugin)
   .settings(addCompilerPlugin("org.typelevel" %% "kind-projector" % "0.10.3"))
   .settings(addCompilerPlugin("org.scalamacros" %% "paradise" % "2.1.1" cross CrossVersion.full))
-  .settings(addCompilerPlugin("com.github.ghik" % "silencer-plugin" % "1.6.0" cross CrossVersion.full))
+  .settings(addCompilerPlugin("com.github.ghik" % "silencer-plugin" % "1.7.0" cross CrossVersion.full))
```
[the remark this answers is not in the document]
Comment (rajanimohan22) [reply build.sbt:52]: Should be fine, I will use that.

File: project/plugins.sbt [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
+resolvers += "HMRC-open-artefacts-maven" at "https://open.artefacts.tax.service.gov.uk/maven2"
+resolvers += Resolver.url("HMRC-open-artefacts-ivy", url("https://open.artefacts.tax.service.gov.uk/ivy2"))(Resolver.ivyStylePatterns)
+resolvers += Resolver.jcenterRepo
 
-resolvers += Resolver.typesafeRepo("releases")
 
-addSbtPlugin("uk.gov.hmrc"               % "sbt-auto-build"     % "2.10.0")
-addSbtPlugin("uk.gov.hmrc"               % "sbt-git-versioning" % "2.1.0")
+addSbtPlugin("uk.gov.hmrc"               % "sbt-auto-build"     % "3.0.0")
 addSbtPlugin("uk.gov.hmrc"               % "sbt-artifactory"    % "1.6.0")
```
[the remark this answers is not in the document]
Comment (rajanimohan22) [reply project/plugins.sbt:10]: Yes, should be removed

File: build.sbt (hunk 2 above)
Comment (rajanimohan22) [reply build.sbt:18]: Yes same as the above one.

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/actions/SessionDataActionBase.scala [hunk 5]
```diff
@@ -46,7 +46,7 @@ trait SessionDataActionBase[R[_] <: Request[_], P[_] <: Request[_]] extends Acti
   ): Future[Either[Result, P[A]]] = {
     implicit val hc: HeaderCarrier =
       HeaderCarrierConverter
-        .fromHeadersAndSession(request.headers, Some(request.session))
+        .fromRequestAndSession(request.withTag("", ""), request.session)
```
[the remark this answers is not in the document]
Comment (rajanimohan22) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/actions/SessionDataActionBase.scala:49]: Myself wondering, how did this come? Let me check.

File: project/plugins.sbt (hunk 4 above)
[the remark this answers is not in the document]
Comment (rajanimohan22) [reply project/plugins.sbt]: Its Funny, I have suggested you the same a week ago about plugins and I myself missed it. There were too many UNDO things, I missed them.

File: build.sbt [hunk 6]
```diff
@@ -15,16 +14,16 @@ resolvers += Resolver.bintrayRepo("hmrc", "releases")
...
       Wart.DefaultArguments,
       Wart.ImplicitConversion,
       Wart.ImplicitParameter,
       Wart.Nothing,
       Wart.Overloading,
       Wart.ToString
     ),
-    wartremoverExcluded in (Compile, compile) ++=
-      routes.in(Compile).value ++
+    wartremover.WartRemover.autoImport.wartremoverExcluded ++=
```
Comment (rambabu-posa) [inline build.sbt:25]: Do we need to do like this?
`(Compile / compile / wartremoverExcluded)`

Comment (rajanimohan22) [reply build.sbt:25]: Yes thats the standard now.

Comment (rajanimohan22) [reply build.sbt:25]: Another example Test / compile for test compilation exclusions

Comment (rajanimohan22) [reply build.sbt:25]: Oops, there is one still not done at Line 29: Done it now.

File: project/AppDependencies.scala [hunk 7]
```diff
@@ -6,22 +6,23 @@ import sbt._
...
   val compile = Seq(
-    "uk.gov.hmrc"                %% "govuk-template"             % "5.58.0-play-26",
-    "uk.gov.hmrc"                %% "play-ui"                    % "8.15.0-play-26",
-    "uk.gov.hmrc"                %% "bootstrap-frontend-play-26" % "3.0.0",
-    "uk.gov.hmrc"                %% "mongo-caching"              % "6.15.0-play-26",
-    "uk.gov.hmrc"                %% "domain"                     % "5.10.0-play-26",
-    "uk.gov.hmrc"                %% "play-language"              % "4.5.0-play-26",
+    "uk.gov.hmrc"                %% "govuk-template"             % "5.66.0-play-26",
+    "uk.gov.hmrc"                %% "play-ui"                    % "9.4.0-play-26",
+    "uk.gov.hmrc"                %% "bootstrap-frontend-play-26" % "4.2.0",
```
Comment (calinursaciuc) [inline project/AppDependencies.scala:11]: 5.3.0 for bootstrap

**Review by calinursaciuc — APPROVED** [review]

**Pull request merged**